internal/pkg/agent/cmd/run.go

```diff
@@ -275,6 +283,24 @@ func run(override cfgOverrider, testingMode bool, fleetInitTimeout time.Duration
 	}
 	defer control.Stop()
 
+	// create symlink from /run/elastic-agent.sock to `paths.ControlSocket()` when running as root
+	// this provides backwards compatibility as the control socket was moved with the addition of --unprivileged
+	// option during installation
+	if isRoot && paths.RunningInstalled() && paths.ControlSocketRunSymlink != "" {
+		socketPath := strings.TrimPrefix(paths.ControlSocket(), "unix://")
+		socketLog := controlLog.With("path", socketPath).With("link", paths.ControlSocketRunSymlink)
+		_ = os.Remove(paths.ControlSocketRunSymlink) // ensure it doesn't exist before creating the symlink
```

Comment:
[Blocker]
If it fails for some other reason that it does not exist?

Reply:
I updated this to log if it gets another error other than not existing.